src/sidebar/store/modules/drafts.ts

```diff
@@ -105,7 +105,10 @@ function createDraft(annotation: AnnotationID, changes: DraftChanges) {
  * An empty draft has no text and no reference tags.
  */
 function deleteNewAndEmptyDrafts() {
-  return (dispatch: Dispatch, getState: () => { drafts: State }) => {
+  return (
+    dispatch: ThunkDispatch<{ drafts: State }, void, any>,
```

Comment:
The `any` represents the actions that `dispatch` can be called with. It's either a regular action, or the return type of an async thunk callback. In this case we need both.

I could not define an action type that didn't make one of the `dispatch(...)` calls trigger a type error, so I defined it as ´any´ for now, but we might want to revisit this.

For references, this is `ThunkDispatch` type definition and docs:

```ts
/**
 * The dispatch method as modified by React-Thunk; overloaded so that you can
 * dispatch:
 *   - standard (object) actions: `dispatch()` returns the action itself
 *   - thunk actions: `dispatch()` returns the thunk's return value
 *
 * @template State The redux state
 * @template ExtraThunkArg The extra argument passed to the inner function of
 * thunks (if specified when setting up the Thunk middleware)
 * @template BasicAction The (non-thunk) actions that can be dispatched.
 */
interface ThunkDispatch<State, ExtraThunkArg, BasicAction extends Action> {
    /** Accepts a thunk function, runs it, and returns whatever the thunk itself returns */
    <ReturnType>(thunkAction: ThunkAction<ReturnType, State, ExtraThunkArg, BasicAction>): ReturnType;
    /** Accepts a standard action object, and returns that action object */
    <Action extends BasicAction>(action: Action): Action;
    /** A union of the other two overloads for TS inference purposes */
    <ReturnType, Action extends BasicAction>(action: Action | ThunkAction<ReturnType, State, ExtraThunkArg, BasicAction>): Action | ReturnType;
}
```

Reply:
I think this is pragmatic for now.